include/fi_util.h

```diff
@@ -400,13 +402,25 @@ struct util_cmap_handle {
 struct util_cmap_peer {
 	struct util_cmap_handle *handle;
 	struct dlist_entry entry;
-	size_t addrlen;
 	uint8_t addr[];
 };
 
-typedef void (*ofi_cmap_free_handle_func)(void *arg);
+typedef struct util_cmap_handle* (*ofi_cmap_alloc_handle_func)(void);
+typedef void (*ofi_cmap_handle_func)(struct util_cmap_handle *handle);
+typedef int (*ofi_cmap_connect_func)(struct util_ep *cmap,
+				     struct util_cmap_handle *handle,
+				     fi_addr_t fi_addr);
+
+struct util_cmap_attr {
+	void 				*name;
+	ofi_cmap_alloc_handle_func 	alloc;
+	ofi_cmap_handle_func 		close;
+	ofi_cmap_handle_func 		free;
```

Comment:
Sorry. I should have been more clear. The close call closes the resources which automatically disconnects the connection. We don't explicitly disconnect.